playground/workflows/src/index.ts

```diff
@@ -0,0 +1,60 @@
+import { WorkflowEntrypoint } from 'cloudflare:workers';
+import type { WorkflowEvent, WorkflowStep } from 'cloudflare:workers';
+
+interface Env {
+	MY_WORKFLOW: Workflow;
+}
+
+export class MyWorkflow extends WorkflowEntrypoint<Env> {
+	override async run(event: WorkflowEvent<Params>, step: WorkflowStep) {
+		await step.do('first step', async () => {
+			return {
+				output: 'First step result',
+			};
+		});
+
+		await step.sleep('sleep', '1 second');
+
+		await step.do('second step', async () => {
+			return {
+				output: 'Second step result',
+			};
+		});
+	}
+}
+
+export default {
+	async fetch(request, env) {
+		const url = new URL(request.url);
+		const id = url.searchParams.get('id');
+
+		if (url.pathname === '/create') {
+			let instance: WorkflowInstance;
+
+			if (id === null) {
+				instance = await env.MY_WORKFLOW.create();
+			} else {
+				instance = await env.MY_WORKFLOW.create({ id });
+			}
+
+			return Response.json({
+				id: instance.id,
+				status: await instance.status(),
+			});
+		} else if (url.pathname === '/get') {
```

Comment:
Nit: given the early returns I think that there's room to simplify this function by removing the various `else` statements:
![Screenshot 2024-12-18 at 17 02 59](https://github.com/user-attachments/assets/3e2913d0-e383-4cec-9bc5-45868cc1e051)

I find the version without the multiple `else`s easier to read, but this is really minor and subjective, so if you prefer your version I'm 1000% cool with that, I just figured I could offer an alternative 🙂

Reply:
d5aae06